rids/rids.py

```diff
@@ -15,113 +15,111 @@
 # limitations under the License.
 
 
+
 """
 Main entry point for RIDS, a Remote Intrusion Detection System.
 """
 
-
 import ipaddress
 import json
-import os
-import subprocess
-import sys
-import urllib.request
+import logging
+import threading
+import urllib
+import queue
 
 from absl import app
 from absl import flags
 
+from rids import config
 from rids import network_capture
+from rids.iocs import iocs
+from rids import rules
 
 FLAGS = flags.FLAGS
 flags.DEFINE_string('host_ip', None,
                     'The IP of the host process, for ignoring direct requests')
-flags.DEFINE_string('config_path', 'config.json',
-                    'Path where IOC configuration can be found; uses config.json by default')
-
-
-def retrieve_bad_ips():
-  bad_ips = {}
-  with open(FLAGS.config_path, 'r') as f:
-    config = json.load(f)
-    for source in config['ioc_sources']:
-      # Download a fresh version of each IOC source in the config.
-      with urllib.request.urlopen(source['url']) as ioc_data:
-        # TODO check format of ioc source, branch to different parsers
-        # For now, the only source format is newline-seprated bad IPv4 addresses
-        # TODO perform parsing in a separate function from config handling
-        for line in ioc_data.readlines():
-          line = line.strip().decode('utf-8')
-          if not line: continue
-          bad_ip = str(ipaddress.ip_address(line))
-          if bad_ip not in bad_ips:
-            bad_ips[bad_ip] = [source['name']]
-          else:
-            bad_ips[bad_ip].append(source['name'])
-  return bad_ips
-
-
-def get_external_ip():
-  with urllib.request.urlopen('https://ipinfo.io/ip') as my_ip:
-    return my_ip.read().decode("utf-8").strip()
+flags.DEFINE_string('config_path', '/etc/rids/config.json',
+                    'file path indicating where IOC configuration can be found')
+flags.DEFINE_string('eventlog_path', '/var/rids/events.log',
+                    'file path indicating where to append new event logs')
 
 
 def main(argv):
+  """Main entry point of the app.  Also bound to CLI `rids` by setuptools."""
   if len(argv) > 1:
     raise app.UsageError('Too many command-line arguments.')
 
+  logging.basicConfig(level=logging.DEBUG,
+                      format='%(asctime)s %(levelname)-8s %(message)s',
+                      datefmt='%m-%d %H:%M',
+                      filename=FLAGS.eventlog_path,
+                      filemode='a')
+
+  cfg = config.ParseFile(FLAGS.config_path)
```

Comment:
Could you make the `ParseFile` return a `RuleSet`?
Otherwise you're exposing the user to the format details.

Reply:
I had in mind that we would probably have other uses for the config.json contents, and it didn't make as much sense to have a "parse config file" method return a ruleset, I intended it to just read and validate (it doesn't validate anything yet).

I can fold that into one "fetch IOCs" call that takes a config, and just read the file contents in main, so that we don't need a config.py anymore.  It would reduce the next four lines so that main() doesn't have to be merging rule sets.